### ISSUE TITLE
Error loading glXCreateContextAttribsARB function on first replay

### ISSUE THREAD

**Issue by darksylinc** [baldurk/renderdoc#1065]
## Description
When on Linux (using built from source v1.1 commit https://github.com/baldurk/renderdoc/commit/c7d5aa863e5f13430b2230e5048d253d342d1a18) replay after capture fails.

However if I later open it manually for replay, it works

## Repro steps
Capture any Linux GL program
Press F12 to capture
Finish the program.

RenderDoc will fail. Log:
```
RDOC 010431: [18:45:32]             core.cpp( 300) - Log     - RenderDoc v1.1 64-bit Release (c7d5aa863e5f13430b2230e5048d253d342d1a18) loaded in replay application
QTRD 010431: [18:45:32]       qrenderdoc.cpp(  88) - Log     - QRenderDoc initialising.
RDOC 010502: [18:45:49]             core.cpp( 300) - Log     - RenderDoc v1.1 64-bit Release (c7d5aa863e5f13430b2230e5048d253d342d1a18) capturing application
RDOC 010502: [18:45:49]   posix_libentry.cpp(  59) - Log     - Using delay for debugger 0
RDOC 010502: [18:45:49]   posix_libentry.cpp(  69) - Log     - Loading into /home/matias/Projects/ColibriGui/bin/Debug/ColibriGui
RDOC 010502: [18:45:49]         gl_hooks.cpp( 168) - Log     - Registering OpenGL hooks
RDOC 010502: [18:45:49]        glx_hooks.cpp( 671) - Log     - Registering GLX hooks
RDOC 010502: [18:45:49]         vk_layer.cpp(  53) - Log     - Registering Vulkan hooks
RDOC 010431: [18:45:49]   target_control.cpp( 512) - Log     - Got remote handshake: ColibriGui [10502]
RDOC 010502: [18:45:49]        gl_driver.cpp( 935) - Log     - Compatibility context 00000000012B8D70 created without attribs, sharing with context 0000000000000000
RDOC 010502: [18:45:49]        gl_common.cpp( 457) - Log     - Checking enabled extensions, running as OpenGL 4.4
RDOC 010502: [18:45:49]      gl_emulated.cpp(2110) - Log     - Emulating EXT_direct_state_access
RDOC 010502: [18:45:49]        gl_driver.cpp( 935) - Log     - Compatibility context 0000000001773AC0 created without attribs, sharing with context 0000000000000000
RDOC 010502: [18:45:49]        gl_common.cpp( 457) - Log     - Checking enabled extensions, running as OpenGL 4.4
RDOC 010502: [18:45:49]      gl_emulated.cpp(2110) - Log     - Emulating EXT_direct_state_access
RDOC 010502: [18:45:50]        gl_driver.cpp( 935) - Log     - Core context 00000000016EE060 created with attribs, sharing with context 0000000000000000
RDOC 010502: [18:45:50]        gl_common.cpp( 457) - Log     - Checking enabled extensions, running as OpenGL 4.5
RDOC 010502: [18:45:50]      gl_emulated.cpp(2110) - Log     - Emulating EXT_direct_state_access
RDOC 010502: [18:45:50]        gl_common.cpp( 490) - Log     - Vendor checks for 45 (X.Org / Radeon RX 560 Series (POLARIS11, DRM 3.23.0, 4.16.3, LLVM 7.0.0) / 4.5 (Core Profile) Mesa 18.2.0-devel (git-71add09e79))
RDOC 010502: [18:45:50]        gl_common.cpp( 733) - Warning - FBOs are shared on this implementation
RDOC 010431: [18:45:50]   target_control.cpp( 720) - Log     - Used API: OpenGL (Not presenting & supported)
RDOC 010431: [18:45:51]   target_control.cpp( 720) - Log     - Used API: OpenGL (Presenting & supported)
RDOC 010502: [18:45:53]        gl_driver.cpp(1756) - Log     - Starting capture, frame 63
RDOC 010502: [18:45:53]        gl_driver.cpp(1774) - Log     - Finished capture, Frame 64
RDOC 010502: [18:45:53]     gl_initstate.cpp(1551) - Log     - Getting tex image for 125
RDOC 010502: [18:45:53]     gl_initstate.cpp(1551) - Log     - Getting tex image for 146
RDOC 010502: [18:45:53]          rdcfile.cpp(1084) - Log     - Finishing write to section 1 (renderdoc/internal/framecapture). Compressed from 206242560 bytes to 924936
RDOC 010502: [18:45:53]             core.cpp(1092) - Log     - Written to disk: /tmp/RenderDoc/ColibriGui_2018.08.04_21.45.49_frame63.rdc
RDOC 010502: [18:45:53]          rdcfile.cpp( 239) - Log     - Opening RDCFile /tmp/RenderDoc/ColibriGui_2018.08.04_21.45.49_frame63.rdc
RDOC 010431: [18:45:53]   target_control.cpp( 673) - Log     - Got a new capture: 0 (time 1533419153) 49105 byte thumbnail
RDOC 010502: [18:45:54]             core.cpp( 353) - Log     - 'Leaking' unretrieved capture /tmp/RenderDoc/ColibriGui_2018.08.04_21.45.49_frame63.rdc
RDOC 010431: [18:45:54]          rdcfile.cpp( 239) - Log     - Opening RDCFile /tmp/RenderDoc/ColibriGui_2018.08.04_21.45.49_frame63.rdc
RDOC 010431: [18:45:54]          rdcfile.cpp( 239) - Log     - Opening RDCFile /tmp/RenderDoc/ColibriGui_2018.08.04_21.45.49_frame63.rdc
RDOC 010431: [18:45:54]     glx_platform.cpp( 467) - Warning - Unable to load 'glXCreateContextAttribsARB'
RDOC 010431: [18:45:54]        gl_replay.cpp(3377) - Error   - Couldn't find required GLX function addresses
RDOC 010431: [18:45:54] replay_controller.cpp(1873) - Error   - Couldn't create a replay device :(.
RDOC 010431: [18:45:54] replay_controller.cpp( 188) - Log     - Shutting down replay renderer
```

It would seem RenderDoc messed up some lib loading during capture hence it cannot open the replay functionality.

If I now go to Open -> Open Capture and open the rdc file in /tmp/RenderDoc it opens up fine.
I haven't yet debugged further.

Btw it was working fine in 0.9x

I'm using Mesa:
```
OpenGL vendor string: X.Org
OpenGL renderer string: Radeon RX 560 Series (POLARIS11, DRM 3.23.0, 4.16.3, LLVM 7.0.0)
OpenGL core profile version string: 4.5 (Core Profile) Mesa 18.2.0-devel (git-71add09e79)
OpenGL core profile shading language version string: 4.50
```

## Environment

Xubuntu 18.04

Once I have more info I will post it here

**Comment by baldurk** [comment]
Can you check with the latest code? I believe this was fixed in 39258daff7d308bbfaa174ff20cfb39448e28112

**Comment by darksylinc** [comment]
OH GOD WHAT THE H---!?

That's a fast reply!!! You didn't even give me time to grab some tea! (literally)

I tested latest. It works. Closing ticket.
